core/trino-main/src/main/java/io/trino/execution/scheduler/EventDrivenFaultTolerantQueryScheduler.java

```diff
@@ -1404,6 +1411,7 @@ private static class StageExecution
         private final int schedulingPriority;
         private final DynamicFilterService dynamicFilterService;
         private final long[] outputDataSize;
+        private volatile boolean producedSomeOutputData;
```

Comment:
We can just compare sizes of `remainingPartitions` and `partitions`?

Reply:
Also this doesn't seem to need to be volatile right? Won't be accessed by multiple threads?

Reply:
Yeah - should work too - for this purpose - good point

Reply:
updated